lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -988,6 +988,188 @@ OpFoldResult ICmpOp::fold(ArrayRef<Attribute> constants) {
   return {};
 }
 
+// Given a range of operands, computes the number of matching prefix and
+// suffix elements. This does not perform cross-element matching.
+static std::pair<size_t, size_t>
+computeCommonPrefixAndSuffix(const OperandRange &a, const OperandRange &b) {
```

Comment:
Would it be possible to implement this one direction and more generically, e.g. something like:

```
template<typename Range>
static size_t computeCommonPrefix(const Range &a, const Range &b) {
...
}
```

and then use something like:

```
auto commonPrefix = computeCommonPrefix(lhsOperands, rhsOperands);
auto commonSuffix = computeCommonPrefix(llvm::reverse(lhsOperands), llvm::reverse(rhsOperands));
```

The `llvm::reverse` algorithm is defined in STLExtras.h.  The only thing to watch for is when they are operand lists are the same, you'll get commonPrefix==commonSuffix.  In this case, just handle it as an equality compare.

Reply:
Done, i done it as follows to remove the additional corner-case.

```c++
  size_t commonPrefixLength = computeCommonPrefixLength(lhsOperands, rhsOperands);
  size_t commonSuffixLength = computeCommonPrefixLength(
      llvm::reverse(lhsOperands.drop_front(commonPrefixLength)),
      llvm::reverse(rhsOperands.drop_front(commonPrefixLength)));
```